components/collapse/collapse.js

```diff
@@ -0,0 +1,129 @@
+import React, {
+  Component,
+  Children,
+  cloneElement,
+  PropTypes,
+} from 'react';
+
+import { getClassNamesWithMods } from '../_helpers';
+import CollapseItem from './collapseItem';
+
+const getNormalizedActiveKey = ({ activeKey = [] }) => {
+  return activeKey instanceof Array ? activeKey : [activeKey];
+};
+
+class Collapse extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeKey: getNormalizedActiveKey(props),
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if ('activeKey' in nextProps) {
+      this.setState({
+        activeKey: getNormalizedActiveKey(nextProps),
+      });
+    }
+  }
+
+  handleItemClick = (e, key) => {
+    const { isAccordion } = this.props;
+    let { activeKey } = this.state;
+
+    if (isAccordion) {
+      activeKey = activeKey[0] === key ? [] : [key];
+    } else {
+      activeKey = [...activeKey];
+      const index = activeKey.indexOf(key);
+
+      if (index >= 0) {
+        activeKey.splice(index, 1);
+      } else {
+        activeKey.push(key);
+      }
+    }
+    this.setActiveKey(activeKey);
+  }
+
+  setActiveKey(activeKey) {
+    if (!('activeKey' in this.props)) {
```

Comment:
when developer will not provide it, then component will use own state, in case, when developer provide `activeKey`(from redux store for example), then collapse will update the state based on props